parlai/crowdsourcing/tasks/qa_data_collection/demo_worlds.py

```diff
@@ -0,0 +1,121 @@
+        super().__init__(opt, mturk_agent)
+        self.opt = opt
+
+    def parley(self):
+        self.mturk_agent.agent_id = "Onboarding Agent"
+        self.mturk_agent.observe({"id": "System", "text": "Welcome onboard!"})
+        x = self.mturk_agent.act(timeout=self.opt["turn_timeout"])
+        self.mturk_agent.observe(
+            {
+                "id": "System",
+                "text": "Thank you for your input! Please wait while "
+                "we match you with another worker...",
+                "episode_done": True,
+            }
+        )
+        self.episodeDone = True
+
+
+class QADataCollectionWorld(MTurkTaskWorld):
+    """
+    World for recording a turker's question and answer given a context.
+    Assumes the context is a random context from a given task, e.g. from SQuAD, CBT,
+    etc.
+    """
+
+    collector_agent_id = 'QA Collector'
+
+    def __init__(self, opt, agent):
+        self.dataloader = opt["dataloader"]
+        self.agent = agent
+        self.agent.agent_id = "QA Agent"
+        self.episodeDone = False
+        self.turn_index = -1
+        self.context = None
+        self.question = None
+        self.answer = None
+        self.opt = opt
+
+    def parley(self):
+        # Each turn starts from the QA Collector agent
+        self.turn_index = (self.turn_index + 1) % 2
+        ad = {'episode_done': False}
+        ad['id'] = self.__class__.collector_agent_id
+
+
+        if self.turn_index == 0:
+            # At the first turn, the QA Collector agent provides the context
+            # and prompts the turker to ask a question regarding the context
+
+            # Get context from dataloader
+            passage = self.dataloader.act()
+            self.context = passage['text']
+            ad['passage'] = passage['text']
+
+            # Wrap the context with a prompt telling the turker what to do next
+            ad['text'] = (
+                'Please provide a question given the passage.'
+            )
+            self.agent.observe(validate(ad))
+            self.question = self.agent.act(timeout=self.opt["turn_timeout"])
+            # Can log the turker's question here
+
+        if self.turn_index == 1:
+            # At the second turn, the QA Collector collects the turker's
+            # question from the first turn, and then prompts the
+            # turker to provide the answer
+
+            # A prompt telling the turker what to do next
+            ad['text'] = 'Thanks. And what is the answer to your question?'
+
+            # ad['episode_done'] = True  # end of episode
+
+            self.agent.observe(validate(ad))
+            self.answer = self.agent.act(timeout=self.opt["turn_timeout"])
+
+            self.episodeDone = True
+
+    def prep_save_data(self, agent):
+        """Process and return any additional data from this world you may want to store"""
+        return {"example_key": "example_value"}
+
+    def episode_done(self):
```

Comment:
This method and `.shutdown()` can be removed to make use of the superclass's methods, which are the same here